Foundation/NSFileManager.swift

```diff
@@ -712,9 +712,15 @@ extension FileManager {
 }
 
 extension FileManager {
-    open var homeDirectoryForCurrentUser: URL { NSUnimplemented() }
+    open var homeDirectoryForCurrentUser: URL {
+        return homeDirectory(forUser: CFCopyUserName().takeRetainedValue()._swiftObject)!
```

Comment:
Looking at CF, it seems the only way for this to totally fail is if even `$HOME` is unset. Perhaps that's an environment error and asserting is the right answer in that case.